sympy/core/expr.py

```diff
@@ -491,6 +495,18 @@ def is_constant(self, *wrt, **flags):
         True
         """
 
+        def check_denominator_zeros(expression):
+
+            from ..solvers import solveset
+            from ..solvers.solvers import denoms
+
+            for den in denoms(expression):
+                if (den.is_zero or
+                    any(solveset(den, symb) is not S.EmptySet
```

Comment:
Sorry, I don't understand, how I can catch that 1/(cos(x) ** 2) - (tan(x)) ** 2, (x - 1)/(1 - x) are not constants without solving equation denominator = 0 or without domains of expression.